fiftyone/core/odm/database.py

```diff
@@ -1615,4 +1642,8 @@ def _delete_run_results(conn, result_ids):
     conn.fs.chunks.delete_many({"files_id": {"$in": result_ids}})
 
 
+def _delete_stores(conn, dataset_ids):
+    conn.execution_store.delete_many({"dataset_id": {"$in": dataset_ids}})
```

Comment:
Shouldn't this use `ExecutionStoreService` ? Or if not what is the reason to deviate from the pattern in delete()

Reply:
This module is like a "break glass in case of emergency" module that can cleanup stuff when the builtin operations fail. For example, it has a `delete_dataset()` method that will delete all remnants of a dataset that can't be loaded via `fo.load_dataset()`.

So `drop_orphan_stores()` is implemented without relying on `ExecutionStoreService` so that if, hypothetically, `ExecutionStoreService` required the dataset to actually exist (which it currently does not, but could in the future), this method would still be able to do the garbage collection by going straight to the database.